### PR TITLE
Update instance questions and points on assessment instance insert

### PR DESCRIPTION
**Pull request by jonatanschroeder**
In some very narrow scenarios it is plausible for a student to create an assessment instance and not open it right away (e.g., in case of network interruption between the new instance POST operation and the GET that first opens the assessment instance). For Homeworks, this could cause a scenario that does not have instance questions or points.

This PR causes the update operation (which creates instance questions and updates points) to always be executed on assessment instance creation, regardless of assessment type. This will have the minor inconvenience that the update operation will run twice on start (once when it's created during POST, once in the redirected GET), but the second update will in essence be a no-op, and as a bonus the assessment instance data will be retained from the middleware instead of being retrieved from a new query.

I also changed the related admin queries, since they no longer need to call update now. Also, `assessment_instances_insert` had an out parameter for new instance question IDs which was never used, so I removed it as well.

### CODE REVIEW

File: apps/prairielearn/src/sprocs/assessment_instances_regrade.sql [hunk 1]
```diff
@@ -31,8 +30,7 @@ BEGIN
     -- first update Homeworks
     updated := FALSE;
     IF assessment_type = 'Homework' THEN
-        SELECT *
-        INTO updated, new_instance_question_ids
+        SELECT updated INTO updated
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/sprocs/assessment_instances_regrade.sql:33]: This is not directly related, but I detected this as unused while removing the other parameter with the same name, so I removed it as well.

**Review by mwest1066 — APPROVED** [review]

File: apps/prairielearn/src/sprocs/assessment_instances_insert.sql [hunk 2]
```diff
@@ -8,8 +8,7 @@ CREATE FUNCTION
         IN time_limit_min integer DEFAULT NULL,
         IN date timestamptz DEFAULT NULL,
         IN client_fingerprint_id bigint DEFAULT NULL,
-        OUT assessment_instance_id bigint,
-        OUT new_instance_question_ids bigint[]
```
Comment (nwalters512) [inline apps/prairielearn/src/sprocs/assessment_instances_insert.sql:12]: AFAICT this column is never used when returned from `assessment_instances_update`; we might be able to remove it from that for a small bit of cleanup? I'm also happy to merge this PR as-is though.

**Review by nwalters512 — APPROVED** [review]

File: apps/prairielearn/src/sprocs/assessment_instances_insert.sql (hunk 2 above)
Comment (jonatanschroeder) [reply apps/prairielearn/src/sprocs/assessment_instances_insert.sql:12]: I thought of that and even drafted that change, but just in case there is any plan in the near future to e.g., highlight the new instance questions this may become useful. The computation of new instances is used internally in assessment_instances_update, but indeed not outside, so one could convert the parameter into a local variable. I'll leave that for a future PR.
